src/lib.rs

```diff
@@ -43,6 +43,9 @@ pub mod poll;
 
 pub mod net;
 
+#[cfg(not(any(target_os = "android", target_os = "haiku")))]
```

Comment:
We prefer to avoid blocklists with `#cfg`s, can you instead whitelist all appropriate platforms?